### PR TITLE
Improve Ansible version checking

### PR DESCRIPTION
**Pull request by BR0kEN-**
This is a bit changed vision of base requirement checker. @PavelLoparev, take a look at this.

### CODE REVIEW

File: requirements.sh [hunk 1]
```diff
@@ -1,17 +1,22 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (BR0kEN-) [inline requirements.sh]: Better to take command from environment, rather than hardcode path to executable.

File: requirements.sh [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
 
-# Determine whether ansible is installed or not.
-info="Please install ansible 1.9.*. Visit http://docs.cibox.tools/en/latest/Requirements for more information."
-is_ansible_installed=false
-ansible --version >/dev/null 2>&1 && { is_ansible_installed=true; }
-if [ "$is_ansible_installed" = false ]; then
-  echo ${info}
+help="Visit http://docs.cibox.tools/en/latest/Requirements for more information."
+ansible=$(which ansible)
```
Comment (BR0kEN-) [inline requirements.sh]: This was made to not execute `ansible --version` twice.

File: requirements.sh [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
+
+if [ -z ${ansible} ]; then
+  echo "Ansible not installed. Further interaction is not possible. ${help}"
   exit 1
```
Comment (BR0kEN-) [inline requirements.sh]: Exit codes must always be different for debugging purposes. This is a basis of shell scripting.

File: requirements.sh [hunk 4]
```diff
@@ -1,17 +1,22 @@
...
-  exit 1
+ansible_version_required="1.9.4"
+# Resulting output of "ansible --version" will be something like this:
+# ansible 2.1.2.0
+#   config file =
+#   configured module search path = Default w/o overrides
+# Use the second column from first row.
+ansible_version_current=$(${ansible} --version | head -1 | awk '{print $2}')
+
+if ! echo ${ansible_version_current} | grep ${ansible_version_required} >/dev/null; then
```
Comment (BR0kEN-) [inline requirements.sh]: More simple check without unneeded variables.

File: requirements.sh (hunk 1 above)
Comment (PavelLoparev) [inline requirements.sh]: If so then we need to change throughout in the cibox *.sh files (provision, repository, somthing else)

**Comment by podarok** [comment]
@BR0kEN-  needs small conflicts resolving

**Comment by BR0kEN-** [comment]
I've started to resolve conflict and have couple questions now:

After execution of `pip show Jinja2` I had:

```
pip show Jinja2
You are using pip version 8.1.2, however version 9.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

That's was not a problem - PIP updated. Then, I've executed `pip show Jinja2` and get nothing. Then execution of `echo $?` gave `1`.

Different Ansible versions implements different way to distribute the vendor modules. Need to think a bit on this.

**Comment by BR0kEN-** [comment]
@podarok, @PavelLoparev, so for now I've removed Jinja2 checker since Ansible distribution could have a lot of ways to have it installed.

What we should do - it's write a playbook and check some Jinja2 feature, that was added in `2.8+` there. In this way we can ensure that Ansible environment has correct Jinja2 version.

**Comment by BR0kEN-** [comment]
In my case Jinja2 installed automatically with Ansible, using `Homebrew` since I'm MacOS user. Also it could be done using `easy_install`, `pip`or from sources.

**Comment by PavelLoparev** [comment]
@BR0kEN- You're right. It's complicated task to check jinja2 vrsion. It's better to have ansible task for checking jinja2 version. If ansible started it means it passed version check.

**Comment by podarok** [comment]
Or we could upgrade our playbooks for ANsible 2

**Comment by danylevskyi** [comment]
Frankly speaking, I think that the quickest and straightforward way to solve this problem is to convert the scripts to support Ansible 2.

We do not need converting ALL roles to Ansible 2. Only the code which generates projects. I believe there are only a few lines of code need to be changed.

**Comment by BR0kEN-** [comment]
Need to merge this because some guys might become annoyed, if they'll take the `master` branch and start the script with installed Jinja2 not via `pip`.

**Comment by BR0kEN-** [comment]
@danylevskyi, you will surprised how many needs to be done to support Ansible 2. I've completed this challenge in my fork half a year ago.

**Pull request merged**